src/synthetix/StakingRewards.t.sol

```diff
@@ -270,16 +285,115 @@ contract StakingRewardsTest is Test {
         assertEq(rewards.rewardsDuration(), 70 days);
     }
 
-    function testRevertSetRewardsDurationOnActiveDistribution() public {
-        setupStakingToken(100 * WAD);
-        rewards.stake(100 * WAD);
+    function testChangeRewardDurationOnActiveDistribution(uint256 newDuration) private {
+        uint256 totalStake = 100 * WAD;
+        uint256 totalReward = 123 * WAD;
+        init7DayFarm(totalStake, totalReward);
 
-        setupReward(100 * WAD);
+        skip(1 days);
+
+        rewards.setRewardsDuration(newDuration);
+
+        assertEq(rewards.rewardPerTokenStored(), 1 days * (totalReward / 7 days) * WAD / totalStake);
+        assertEq(rewards.lastUpdateTime(), block.timestamp);
+        assertEq(rewards.rewardRate(), 6 days * (totalReward / 7 days) / newDuration);
```

Comment:
Done

Reply:
Not asking to change, but imo reading the tests with the the explicit days amount (`6 days`, `7 days`, etc..) is much simpler and spares the cognitive conversion you have to do when reading the code with the variables.

As opposed to the source contracts or fuzzing tests, testing specific cases with easy to read specific numeric values is good (in case it does not add much code duplication).

Reply:
I agree but in this case I made the change because the variables in question are involved in a tiny bit of arithmetic (i.e. the 7-1 subtraction) so I can see an argument for replacing the hardcoded numbers with vars. That said I would prefer not to extend that to numbers that are not used in arithmetic as I agree that in that case it's overkill and increases the cognitive load.